gensim/models/word2vec.py

```diff
@@ -474,6 +474,11 @@ def __init__(
                 raise TypeError("You can't pass a generator as the sentences argument. Try an iterator.")
             self.build_vocab(sentences, trim_rule=trim_rule)
             self.train(sentences)
+            
+        else :
```

Comment:
This warning even shows if no `trim_rule` specified – which seems excessive, as there's no hint that the user is confused in that case.